src/shell.js

```diff
@@ -186,7 +186,7 @@ function logExceptionOnExit(e) {
 }
 #endif
 
-#if ENVIRONMENT_MAY_BE_NODE
+#if ENVIRONMENT_MAY_BE_NODE && WASM
```

Comment:
I don't think this has an effect. `WASM` is always true in this part of the compiler - building with `-sWASM=0` is shorthand for `-sWASM2JS`, but it sets `WASM=1` internally (as we do build to wasm, and then follow up with wasm2js).

Unless I'm missing something - you can verify by diffing output with and without this patch. I don't see a difference locally when I do that.

Reply:
Would `!WASM2JS` work in this case?
For my own curiosity, do you know where does it set `WASM=1`?

Reply:
So it looks like `!WASM2JS` doesn't work either 🤔 I guess its value is also unreliable because it's going to change depending on what step of the compilation we're on? Maybe we need another internal flag `NEED_PATH_AND_FS_API` or `USER_PROVIDED_WASM_VALUE` to ensure we only remove the code when we mean to do that, wdyt?